### PR TITLE
fix: Tagging new region only applies to one region [AB#17176]

### PR DESCRIPTION
**Pull request by tbarlow12**
- Canvas Helpers now returns clones of transformed tag arrays
- Canvas now allows CanvasTools to manage selected regions (removed from Canvas state)
- Fixing tag tests from editor page

### CODE REVIEW

File: src/react/components/pages/editorPage/editorPage.test.tsx [hunk 1]
```diff
@@ -413,14 +424,6 @@ describe("Editor Page Component", () => {
...
         });
 
-        function dispatchKeyEvent(key: string, eventType: KeyEventType = KeyEventType.KeyDown) {
-            window.dispatchEvent(new KeyboardEvent(
-                eventType, {
-                    key,
-                },
-            ));
-        }
-
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/editorPage.test.tsx]: Why are you getting rid of this function?

File: src/react/components/pages/editorPage/editorPage.test.tsx [hunk 2]
```diff
@@ -498,6 +501,27 @@ describe("Editor Page Component", () => {
             dispatchKeyEvent("Ctrl+e");
             expect(clearRegions).toBeCalled();
         });
```
Comment (elizabethhalper) [inline src/react/components/pages/editorPage/editorPage.test.tsx]: Make sure you are up to date with the current dev. The Ctrl+e is related to the export provider and I don't know if this test is valid anymore.

Comment (tbarlow12) [reply src/react/components/pages/editorPage/editorPage.test.tsx]: Good call. I'll make the change

Comment (tbarlow12) [reply src/react/components/pages/editorPage/editorPage.test.tsx]: Good call. I'll make the change

File: src/react/components/pages/editorPage/editorPage.test.tsx (hunk 1 above)
Comment (tbarlow12) [reply src/react/components/pages/editorPage/editorPage.test.tsx]: I think I moved it somewhere else... I'll look into it

**Pull request merged**